performance_manager/lib/s3_utils.py

```diff
@@ -0,0 +1,51 @@
+import logging
+import os
+from collections.abc import Iterable
+
+import boto3
+import pandas
+import pyarrow.parquet as pq
+from pyarrow import fs
+
+
+def read_parquet(filename: str) -> pandas.core.frame.DataFrame:
```

Comment:
Can we add a `filters` parameter to this function? https://arrow.apache.org/docs/python/generated/pyarrow.parquet.ParquetDataset.html

I think we will not always want the whole parquet file returned as a data-frame and this would probably be the most efficient way to limit what is returned.